From b0a0c608a1309ae1a5b62d8c0d70772547dee25c Mon Sep 17 00:00:00 2001
From: Vitaly Prosyak <vitaly.prosyak@amd.com>
Date: Thu, 14 Jul 2016 13:57:00 -0400
Subject: [PATCH 0617/1722] drm/amdgpu: add freesync capable immune proeprty

remove declaration of handler for freesync drm property
in based driver code path, since it is used by dal through
the atomic framework.

Change-Id: I202010488a9b3b4fe7dde1623afc71b755b8e73a
Signed-off-by: Vitaly Prosyak <vitaly.prosyak@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_connectors.c | 5 -----
 drivers/gpu/drm/amd/amdgpu/amdgpu_display.c    | 6 ++++++
 drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h       | 2 ++
 3 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_connectors.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_connectors.c
index 5708019..ff0b55a 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_connectors.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_connectors.c
@@ -625,11 +625,6 @@ static int amdgpu_connector_set_property(struct drm_connector *connector,
 		amdgpu_connector_property_change_mode(&amdgpu_encoder->base);
 	}
 
-	if (property == adev->mode_info.freesync_property &&
-	    adev->mode_info.funcs->set_freesync_property)
-		adev->mode_info.funcs->set_freesync_property(connector,
-							     property, val);
-
 	return 0;
 }
 
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_display.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_display.c
index 7630950..5370653 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_display.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_display.c
@@ -665,6 +665,12 @@ int amdgpu_modeset_create_props(struct amdgpu_device *adev)
 			drm_property_create_bool(adev->ddev, 0, "freesync");
 		if (!adev->mode_info.freesync_property)
 			return -ENOMEM;
+		adev->mode_info.freesync_capable_property =
+			drm_property_create_bool(adev->ddev,
+						 DRM_MODE_PROP_IMMUTABLE,
+						 "freesync_capable");
+		if (!adev->mode_info.freesync_capable_property)
+			return -ENOMEM;
 	}
 
 	return 0;
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
index 5e0756b..502a20b 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
@@ -349,6 +349,8 @@ struct amdgpu_mode_info {
 	struct drm_property *dither_property;
 	/* it is used to allow enablement of freesync mode */
 	struct drm_property *freesync_property;
+	/* it is used to know about display capability of freesync mode */
+	struct drm_property *freesync_capable_property;
 	/* hardcoded DFP edid from BIOS */
 	struct edid *bios_hardcoded_edid;
 	int bios_hardcoded_edid_size;
-- 
2.7.4

